File: TypeScript/codewars/The-Feast-of-Many-Beasts.ts
export function feast(beast: string, dish: string): boolean {
	return getFirstAndLastLetter(beast) === getFirstAndLastLetter(dish);
}

function getFirstAndLastLetter(str: string): string {
	const firstLetter = str.charAt(0);
	const lastLetter = str.charAt(str.length - 1);

	return `${firstLetter}${lastLetter}`;
}

File: Csharp/CodeWars/src/Resolutions/TheFeastOfManyBeasts.cs
namespace CodeWars.src.Resolutions
{
    public class TheFeastOfManyBeasts
    {
        public static bool Feast(string beast, string dish) =>
            GetFirstAndLastLetter(beast) == GetFirstAndLastLetter(dish);

        private static string GetFirstAndLastLetter(string str) =>
            $"{str.First()}{str.Last()}";
    }
}
